### PR TITLE
refactor:  packages\ui\src\components\va-affix\VaAffix.vue

### PR DESCRIPTION
**Pull request by LighthouseKeeperYN**
#1397

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-affix/VaAffix.vue [hunk 1]
```diff
@@ -16,168 +16,143 @@
...
-      }
+      if (props.offsetTop === undefined) { return }
 
-      if (target !== window) {
-        const { top } = (target as HTMLElement).getBoundingClientRect()
-        return top + this.offsetTop
+      if (!(target instanceof Window)) {
+        const { top } = target.getBoundingClientRect()
+        console.log(top)
+        console.log(props.offsetTop)
```
Comment (m0ksem) [inline packages/ui/src/components/va-affix/VaAffix.vue:61]:
```suggestion
```

File: packages/ui/src/components/va-affix/VaAffix.vue [hunk 2]
```diff
@@ -16,168 +16,143 @@
...
-    }
-
-    if (!eventName || eventName === 'resize') {
-      handleThrottledEvent(eventName, context)
-    } else if (event && event.target) {
-      const target = this.getTargetElement()
+    const computedClass = computed(() => [{ 'va-affix--affixed': isAffixed }])
+    const computedStyle = computed(() => ({
+      top: state.value.isTopAffixed ? convertToPixels(calculateTop) : null,
+      bottom: state.value.isBottomAffixed ? convertToPixels(calculateBottom) : null,
```
Comment (m0ksem) [inline packages/ui/src/components/va-affix/VaAffix.vue:92]: Better ts typing for CSSProperties type.

```suggestion
      top: state.value.isTopAffixed ? convertToPixels(calculateTop) : undefined,
      bottom: state.value.isBottomAffixed ? convertToPixels(calculateBottom) : undefined,
```

**Review by m0ksem — CHANGES_REQUESTED** [review]

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]
Looks good!
Pay attention to one remark.

File: packages/ui/src/components/va-affix/VaAffix.vue [hunk 3]
```diff
@@ -16,168 +16,141 @@
...
+
+      if (props.offsetBottom === undefined) { return }
 
-      if (target !== window) {
-        const { bottom } = (target as HTMLElement).getBoundingClientRect()
-        const { offsetHeight, clientHeight } = (target as HTMLElement)
+      if (!(target instanceof Window)) {
+        const { bottom } = target.getBoundingClientRect()
+        const { offsetHeight, clientHeight } = target
         const scrollBarHeight = offsetHeight - clientHeight
```
[the remark this answers is not in the document]
Comment (LighthouseKeeperYN) [reply packages/ui/src/components/va-affix/VaAffix.vue]: fixed

**Review by RVitaly1978 — APPROVED** [review]

**Pull request merged**